common/types/yamlUtils.go

```diff
@@ -19,25 +21,114 @@ func ReadYaml(filePath string, primaryKey string, content interface{}, hooks []m
 	// handle proto enums
 	opts := []viper.DecoderConfigOption{
 		func(dc *mapstructure.DecoderConfig) {
-			dc.ErrorUnset = true
 			dc.ErrorUnused = true
 		},
 	}
 	if len(hooks) != 0 {
+		hookFuncs := make([]mapstructure.DecodeHookFunc, 0, len(hooks))
 		for _, h := range hooks {
-			opts = append(opts, viper.DecodeHook(h))
+			hookFuncs = append(hookFuncs, mapstructure.ComposeDecodeHookFunc(h))
 		}
+		opts = append(opts, viper.DecodeHook(mapstructure.ComposeDecodeHookFunc(hookFuncs...)))
 	}
 
-	err := viper.ReadInConfig()
+	err = viper.ReadInConfig()
 	if err != nil {
-		return err
+		return missingFields, err
 	}
 
 	err = viper.GetViper().UnmarshalKey(primaryKey, content, opts...)
 	if err != nil {
-		return err
+		return missingFields, err
 	}
 
-	return nil
+	missingFields = findMissingFields(content)
+	return missingFields, nil
+}
+
+func findMissingFields(content interface{}) []string {
+	var missingFields []string
+
+	expectedFields := make(map[string]bool)
+	contentValue := reflect.ValueOf(content)
+	contentType := contentValue.Type()
+
+	// Ensure content is a struct or a pointer to a struct
+	if contentType.Kind() != reflect.Struct && (contentType.Kind() != reflect.Ptr || contentType.Elem().Kind() != reflect.Struct) {
+		return missingFields
```

Comment:
if calling `findMissingFields()` with non-struct is a serious bug, it needs to be communicated somehow - panic, or log, or return error.

Reply:
please mark the panic as "ok".

Reply:
done. see https://github.com/lavanet/lava/commit/8610fb0f6e29931a011ce961b80ccd7fba66ad98